<a href="https://colab.research.google.com/github/ryonce/Daily-Projects/blob/main/Classification_Tree_Exercise.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Classifcation Tree Models


## Data and Split

In [3]:
# Imports

import pandas as pd 
from sklearn.model_selection import train_test_split 
from sklearn.tree import DecisionTreeClassifier
from sklearn.ensemble import BaggingClassifier
from sklearn.ensemble import RandomForestClassifier

In [2]:
# Import data

df = pd.read_csv('https://docs.google.com/spreadsheets/d/e/2PACX-1vSyzKH3nFAiPSB6OkLTjDwFlTWO_Lw904LoqljKmHbnrh4ROtS0SFF3SHMGz6HSdPeaB4LBC8nE1_rG/pub?output=csv')
df.head()

id diagnosis  radius_mean  texture_mean  perimeter_mean  area_mean  \
0    842302         M        17.99         10.38          122.80     1001.0   
1    842517         M        20.57         17.77          132.90     1326.0   
2  84300903         M        19.69         21.25          130.00     1203.0   
3  84348301         M        11.42         20.38           77.58      386.1   
4  84358402         M        20.29         14.34          135.10     1297.0   

   smoothness_mean  compactness_mean  concavity_mean  concave points_mean  \
0          0.11840           0.27760          0.3001              0.14710   
1          0.08474           0.07864          0.0869              0.07017   
2          0.10960           0.15990          0.1974              0.12790   
3          0.14250           0.28390          0.2414              0.10520   
4          0.10030           0.13280          0.1980              0.10430   

   ...  radius_worst  texture_worst  perimeter_worst  area_worst  \
0  ...         25.38          17.33           184.60      2019.0   
1  ...         24.99          23.41           158.80      1956.0   
2  ...         23.57          25.53           152.50      1709.0   
3  ...         14.91          26.50            98.87       567.7   
4  ...         22.54          16.67           152.20      1575.0   

   smoothness_worst  compactness_worst  concavity_worst  concave points_worst  \
0            0.1622             0.6656           0.7119                0.2654   
1            0.1238             0.1866           0.2416                0.1860   
2            0.1444             0.4245           0.4504                0.2430   
3            0.2098             0.8663           0.6869                0.2575   
4            0.1374             0.2050           0.4000                0.1625   

   symmetry_worst  fractal_dimension_worst  
0          0.4601                  0.11890  
1          0.2750                  0.08902  
2          0.3613                  0.08758  
3          0.6638                  0.17300  
4          0.2364                  0.07678  

[5 rows x 32 columns]

In [4]:
# Assign the M and B for 0 & 1

df['diagnosis'] = df['diagnosis'].replace({'M': 0, 'B': 1,})

In [5]:
# Arrange Data into a Features Matrix and Target Vector.

y = df['diagnosis']
X = df.drop(columns = 'diagnosis')

In [6]:
# Train Test Split

X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=42)

## Decision Classifier

In [7]:
# Name the model and fit it to the data

dec_tree = DecisionTreeClassifier()

dec_tree.fit(X_train, y_train)

DecisionTreeClassifier()

In [8]:
# Calculate classification accuracy

train_score = dec_tree.score(X_train, y_train)
test_score = dec_tree.score(X_test, y_test)

print(f'Train score for Decision Tree: {train_score}')
print(f'Test score for Decision Tree: {test_score}')

Train score for Decision Tree: 1.0
Test score for Decision Tree: 0.9300699300699301


## Bagging Classifier

In [9]:
# Name the model and fit it to the data

bag_tree = BaggingClassifier()

bag_tree.fit(X_train, y_train)

BaggingClassifier()

In [11]:
# Calculate classification accuracy

train_score = bag_tree.score(X_train, y_train)
test_score = bag_tree.score(X_test, y_test)

print(f'Train score for Bagging Tree: {train_score}')
print(f'Test score for Bagging Tree: {test_score}')

Train score for Decision Tree: 1.0
Test score for Decision Tree: 0.9440559440559441


## Random Forests

In [12]:
# Name the model and fit it to the data

rf_tree = RandomForestClassifier()

rf_tree.fit(X_train, y_train)

RandomForestClassifier()

In [14]:
# Calculate classification accuracy

train_score = rf_tree.score(X_train, y_train)
test_score = rf_tree.score(X_test, y_test)

print(f'Train score for Random Forest Tree: {train_score}')
print(f'Test score for Random Forest Tree: {test_score}')

Train score for Random Forest Tree: 1.0
Test score for Random Forest Tree: 0.965034965034965


 - The best model based on the data is the Random Forest Tree model with:
    - Test score of .965